components/organisms/NavBar.tsx

```diff
@@ -1,9 +1,14 @@
-import styled from 'styled-components';
+import React from 'react';
 
-export const MobileNavBar = styled.div`
+import MobileNavBar from './MobileNavbar';
+import DesktopNavbar from './DesktopNavbar';
+import { useViewport } from './ViewportProvider';
 
-`;
+const NavBar = () => {
+  const width = useViewport();
+  const breakpoint = 1200;
 
-const NavBar = () => <div>NAVBAR!</div>;
```

Comment:
Looking at figma I don't feel justified to have multiple components for desktop and mobile layout.

For me it feels like that this could be achieved a bit easier with building desktop first, and adding @media selectors to the styled component definitions. I think it would be easier to maintain too, the navbar implementations seem almost identical duplicates.

What do you guys think ?

Reply:
I do agree. In general it is not a great practice to have components like `LogoAndNameDesktop` and `LogoAndNameMobile` unless their functionality or core presentation value show serious differences. In this case we are talking about only CSS differences.

Reply:
Wait a second, did @mikitovicsmark mean the LogoAnd... components here, or the whole Navbar? 

If the second, I was considering this, and could not think of a way to solve it simply with breakpoints. I agree that up to a point they are similar, but one has a closed menu drawer and a logic to toggle it, even different tools are used etc. Upon some thinking, and some brief research I found this is more feasible to me.

An aside on mobile-first vs desktop first: before the project started, we discussed this, and agreed on this method with Imre. The reason was that Bálint is highly in favor of mobile-first, and while we don't really understand while, we were alright with this way since we don't have any preferences. We, as the Frontend Team 🎉 can agree on desktop first-development, but Bálint may have some comments there :D

Reply:
`mobile-first` is a general best practice in frontend development for more than a decade now. I think the primary benefit is that it makes sure you are always prepared for the worst-case scenario. Also desktop is usually much more complicated compared to a mobile view (except the navbar), meaning, when you style the next level, you will not have to revert a bunch of additional styles that are only relevant on desktop.

Although, I don't think that makes any difference here. You can implement it the same way without additional code with CSS media queries.

Reply:
There's something else that was missed here, but now I remember: I have a plan to move that chevron around, based on which page the user is at. (This way, the state shown on Figma would mean we are on the 'Amiben segíteni tudok' page. Naturally, by clicking on it, a jump will be made to the lower section of the page with the content.) This is another fuctionality that would work on desktop only. I feel way too uncomfortable solving so many issues within the same component, so I think I will go with 2 components + a CSS media query (just as Bálint suggested). 

If anybody feels up for that, I am open to organize a pair-coding sesh to make it the other way though.